django/db/models/expressions.py

```diff
@@ -1156,7 +1156,7 @@ def as_sqlite(self, compiler, connection):
     def as_mysql(self, compiler, connection):
         template = None
         if self.nulls_last:
-            template = 'IF(ISNULL(%(expression)s),1,0), %(expression)s %(ordering)s '
+            template = 'ISNULL(%(expression)s), %(expression)s %(ordering)s '
```

Comment:
Could this use the operator instead of the function? So `%(expression)s IS NULL`. It may need some parentheses round the expression, not sure.